src/main/groovy/com/avioconsulting/mule/linter/model/Application.groovy

```diff
@@ -3,9 +3,36 @@ package com.avioconsulting.mule.linter.model
 class Application {
 
     File applicationPath
+    Map<String,ProjectFile> files = new HashMap<String,ProjectFile>()
 
     Application(File applicationPath) {
         this.applicationPath = applicationPath
+        if( !this.applicationPath.exists()) {
+            throw new FileNotFoundException(applicationPath.absolutePath, "Application directory does not exists.")
+        }
+
+        //Parse out the files necessary for an application.
+        files.put("POM", new File(applicationPath, "pom.xml"))
```

Comment:
What is this going to look like for other files?  We shouldn't just be using a static string here, it would be a constant if it is.

Reply:
my thought would be to have a FileType enum... (but then we'll have to make the projectFile a list)
And as we add more 'files' that have rules, they get added into the constructor code to find and add that file to the map...